### PR TITLE
BGDIINF_SB-1866: Addapted to the new KML micros service

### PR DESCRIPTION
**Pull request by ltshb**
[Test link](https://web-mapviewer.dev.bgdi.ch/feature_bgdiinf_sb-1866_new_kml_svc/index.html)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +66,7 @@
     "babel-eslint": "^10.1.0",
     "caniuse-lite": "^1.0.30001255",
     "chai": "^4.3.4",
+    "cypress": "^7.7.0",
```
[the remark this answers is not in the document]
Comment (ltshb) [reply package.json]: I removed it

File: src/api/files.api.js [hunk 2]
```diff
@@ -1,36 +1,139 @@
...
+        const errorMessage = `Needs a valid KML`
+        log('error', errorMessage)
+        reject(errorMessage)
+    }
+}
+
+function buildKmlForm(kml, reject) {
+    validateKml(kml, reject)
+    const form = new FormData()
+    const blob = new Blob([kml], { type: 'application/vnd.google-earth.kml+xml' })
```
[the remark this answers is not in the document]
Comment (ltshb) [reply src/api/files.api.js:92]: This is handled in the `validateKml()` method in line 90. In this case the method is rejected with an error message.

File: src/api/files.api.js [hunk 3]
```diff
@@ -47,95 +150,62 @@ export const create = (kml) => {
...
-        if (!id) {
-            const errorMessage = `Needs a valid file ID`
-            log('error', errorMessage)
-            reject(errorMessage)
-        }
-        if (!kml || !kml.length) {
-            const errorMessage = `Needs a valid KML`
-            log('error', errorMessage)
-            reject(errorMessage)
-        }
```
[the remark this answers is not in the document]
Comment (ltshb) [reply src/api/files.api.js:65]: This check has been moved in `validateKml()` method.

File: src/api/layers/KMLLayer.class.js [hunk 4]
```diff
@@ -4,18 +4,32 @@ import AbstractLayer from '@/api/layers/AbstractLayer.class'
...
      */
-    constructor(name, opacity, kmlFileUrl) {
+    constructor(name, opacity, kmlFileUrl, fileId = null, adminId = null) {
         super(name, LayerTypes.KML, opacity)
         this.kmlFileUrl = kmlFileUrl
+        this.adminId = adminId
+        if (fileId) {
+            this.fileId = fileId
+        } else {
+            this.fileId = kmlFileUrl.split('/').pop()
```
[the remark this answers is not in the document]
Comment (ltshb) [reply src/api/layers/KMLLayer.class.js:24]: Done

**Review by pakb — APPROVED** [review]

**Pull request merged**